File: Ejercicio_7/Ejercicio_7.py
import random


def generate_new_unique_number(numbers_list):
    number = random.randint(1, 99)
    number_in_list = number in numbers_list
    while number_in_list:
        number = random.randint(1, 99)
        number_in_list = number in numbers_list
    numbers_list.append(number)


# def generate_new_unique_number(numbers_list):
#     number = 0
#     number_in_list = True
#     while number_in_list:
#         number = random.randint(1, 99)
#         number_in_list = number in numbers_list
#     numbers_list.append(number)


def generate_lottery_board():
    lottery_numbers = []

    for i in range(15):
        generate_new_unique_number(lottery_numbers)

    row1 = lottery_numbers[0:5]
    row2 = lottery_numbers[5:10]
    row3 = lottery_numbers[10:15]

    lottery_board = [row1, row2, row3]
    return lottery_board


print("Python")
print(generate_lottery_board())

File: Ejercicio_7/Ejercicio_7.cs
Console.WriteLine("C#");
Console.Write("[");
foreach (var row in Functions.GenerateLotteryBoard())
{
    Console.Write("[");
    Console.Write(String.Join(",", row));
    Console.Write("]");
}
Console.Write("]");

public static class Functions
{
    public static void GenerateNewUniqueNumber(List<int> numbersList)
    {
        int number;
        bool numberInList;
        do
        {
            number = (new Random()).Next(1, 100);
            numberInList = numbersList.Contains(number);
        } while (numberInList);

        numbersList.Add(number);
    }

    public static List<List<int>> GenerateLotteryBoard()
    {
        List<int> lotteryNumbers = new List<int>();

        for (int i = 0; i < 15; i++)
        {
            GenerateNewUniqueNumber(lotteryNumbers);
        }

        List<int> row1 = lotteryNumbers.GetRange(0, 5);
        List<int> row2 = lotteryNumbers.GetRange(5, 5);
        List<int> row3 = lotteryNumbers.GetRange(10, 5);

        List<List<int>> lotteryBoard = new List<List<int>>() { row1, row2, row3 };
        return lotteryBoard;
    }
}